File: java/azure-storage-blob-v8-perfstress/src/main/java/com/azure/storage/blob/v8/perfstress/UploadFromFileTest.java
package com.azure.storage.blob.v8.perfstress;

import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.nio.file.Files;
import java.nio.file.Path;

import com.azure.perfstress.RandomStream;
import com.azure.perfstress.SizeOptions;
import com.azure.storage.blob.v8.perfstress.core.RandomBlobTest;
import com.microsoft.azure.storage.StorageException;

import reactor.core.publisher.Mono;

public class UploadFromFileTest extends RandomBlobTest<SizeOptions> {

    private static Path tempFile;

    public UploadFromFileTest(SizeOptions options) {
        super(options);
    }

        @Override
    public Mono<Void> GlobalSetupAsync() {
        return super.GlobalSetupAsync().then(CreateTempFile());
    }

    @Override
    public Mono<Void> GlobalCleanupAsync() {
        return DeleteTempFile().then(super.GlobalCleanupAsync());
    }

    private Mono<Void> CreateTempFile() {
        try {
            tempFile = Files.createTempFile(null, null);
            
            InputStream inputStream = RandomStream.create(Options.Size);
            OutputStream outputStream = new FileOutputStream(tempFile.toString());
            inputStream.transferTo(outputStream);
            outputStream.close();
            
            return Mono.empty();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    private Mono<Void> DeleteTempFile() {
        try {
            Files.delete(tempFile);
            return Mono.empty();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public void Run() {
        try {
            _cloudBlockBlob.uploadFromFile(tempFile.toString());
        } catch (StorageException | IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public Mono<Void> RunAsync() {
        throw new UnsupportedOperationException();
    }
}

File: net/Azure.Storage.Blobs.V11.PerfStress/UploadFromFileTest.cs
using Azure.Storage.Blobs.PerfStress.Core;
using Microsoft.Azure.Storage.Blob;
using System;
using System.IO;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Storage.Blobs.PerfStress
{
    public class UploadFromFileTest : RandomBlobTest<SizeOptions>
    {
        private static string _tempFile;

        public UploadFromFileTest(SizeOptions options) : base(options)
        { 
        }

        public override async Task GlobalSetupAsync()
        {
            await base.GlobalSetupAsync();

            _tempFile = Path.GetTempFileName();

            using var randomStream = RandomStream.Create(Options.Size);
            using var fileStream = File.OpenWrite(_tempFile);
            await randomStream.CopyToAsync(fileStream);
            fileStream.Close();
        }

        public override async Task GlobalCleanupAsync()
        {
            File.Delete(_tempFile);
            await base.GlobalCleanupAsync();
        }

        public override void Run(CancellationToken cancellationToken)
        {
            CloudBlockBlob.UploadFromFile(_tempFile);
        }

        public override Task RunAsync(CancellationToken cancellationToken)
        {
            return CloudBlockBlob.UploadFromFileAsync(_tempFile);
        }
    }
}
